jsx/Form.js

```diff
@@ -1736,6 +1736,138 @@ class LorisElement extends Component {
   }
 }
 
+/**
+ * Slider Component
+ * React wrapper for a <input type='range'> element.
+ */
+class SliderElement extends React.Component {
+  constructor() {
+    super();
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(e) {
+    // Handles empty, min & max cases.
+    const inputValue = e.target.value
+      ? parseFloat(e.target.value)
+      : this.props.valueMin;
+    let value = inputValue > this.props.valueMax
+      ? this.props.valueMax
+      : inputValue;
+    value = value < this.props.valueMin
+      ? this.props.valueMin
+      : value;
+    this.props.onUserInput(this.props.name, value);
+  }
+
+  render() {
+    let errorMessage = null;
+    let requiredHTML = null;
+    let elementClass = this.props.elementClass;
+    let disabled = this.props.disabled ? 'disabled' : null;
+    let required = this.props.required ? 'required' : null;
+    // Add required asterix
+    if (required) {
+      requiredHTML = <span className='text-danger'>*</span>;
+    }
+    // Add error message
+    if (this.props.errorMessage) {
+      errorMessage = <span>{this.props.errorMessage}</span>;
+      elementClass = this.props.elementClass + ' has-error';
+    }
+
+    return (
+      <div className={elementClass}>
+        <label className={'col-sm-3 control-label'}
+               htmlFor={this.props.id}>
+          {this.props.label}
+          {errorMessage}
+          {requiredHTML}
+        </label>
+        <div className={'col-sm-9'}>
+          <div style={{
+            display: 'flex',
+            flexDirection: 'row',
+            flexWrap: 'wrap',
+            width: '100%',
+          }}>
+            <div style={{
+              display: 'flex',
+              flexDirection: 'column',
+              flexBasis: '100%',
+              maxWidth: '50px',
+              flex: 1,
+            }}>
+              <input
```

Comment:
could we have the value box come after the slider? format/styling wise as well, i think it would be nice to have the slider element take half the space when big (elements can layer when the browser size is small), and float the input box to the left

Reply:
Hey @zaliqarosli, can you expand on why it would be better to have the input on the right. I think it looks more aesthetically pleasing on the left compared to the right. Also when I go with my mouse to type into an input box. I find myself moving the mouse nearest to the left of the input and not the right. No big deal for me if you desire the right though. I just find myself always going to the far left of the input field when clicking on it.

Lastly can you explain more about "slider element take half the space when big (elements can layer when the browser size is small), and float the input box to the left" as I'm not certain when looking at the component when resizing the browser.

I'll include screenshots of current implementation. My thoughts right now are the current implementation saves vertical space when having the input next to the slider and even on the smallest resize of chrome there is a lot to work with for the slider. I would think that's better as typically vertical space is valuable. Interested to read your thoughts.

![example size](https://user-images.githubusercontent.com/16293415/66331635-fe771680-e900-11e9-89ca-df5eeddeb610.png)

Reply:
for your first question, the way the form is set up makes the user interact with the form and its elements from top to bottom, and left to right. so it makes sense that you would automatically go to the left of the field before going to the right. because the range input is the only really important part of this element (the slider should be the main focus of a SliderElement), it should come first. the user should interact with it first before choosing to use the direct number input if they so desire. if we're encouraging users to use the number input first and ignore the slider, than there's no point in adding this new element, and we could use the NumericElement component instead. the value box is a display of the value at the very least, but is also additionally an input option

Reply:
for the second point, when the browser is at full size, i just think the slider is way bigger than is necessary. so, at full size (left example in screenshot), i think the slider can take up half the space, and when not at full size (right example in screenshot), the slider can take up the whole space.

i also think there should be some margin/padding between the value box and the slider. they are very close to each other right now. they should have the same spacing as what's to the left and right of all those input fields.

Reply:
@zaliqarosli I think it's hard to know for certain since we're kind of postulating on the left being preferred. I'll move it to the right because I don't use sliders much to know what makes sense and I never use them to be certain about this situation.

Reply:
I would appreciate it if you could just trust me on that first point.

for the second point, it is true that all the other fields don't need to be as big as they are. if you are not willing to change the size of the slider, then maybe it would be best to give the person implementing the component the option to set the size via a prop.

seeing as the first use case of the slider is the new instrument builder, it would be nice if I could use this component the way I need to use it. thanks for your help!

Reply:
Hey @zaliqarosli I trust you!!  I wasn't trying to dismiss your point about the slider can be too large in width. I'm just trying to clarify everything before modifying the css. I'm working from home today because of a cold with an appointment I have to be at and otherwise I would just clarify everything in person. I'll try adding a prop that manages the max width because then the component can be used in other cases without looking giant.

Reply:
thank you!